libMBIN/Source/Template/NMSTemplate.cs

```diff
@@ -604,6 +601,9 @@ public void SerializeGenericList( BinaryWriter writer, IList list, long listHead
                             SerializeList( writer, (IList) data.Item2, data.Item1, ref listObjects, i + 1, listEnding );
                         }
                     } else {
+                        int alignment2 = data.Item2.GetType().GetCustomAttribute<NMSAttribute>()?.Alignment ?? 0x8;
```

Comment:
Is this change to fix the alignment stuff for strings?
I have actually been working on a more concrete fix for string alignments based on completely changing all
```
[NMS(Size = 0x10)]
public string FieldName
```
instances to
```
public NMSString0x10 FieldName
```
(and same for all the other size strings and fixing the alignment issues this way.
I have it all implemented and working, so I am not sure if this change will be needed/right once that happens, but I am not sure...

Reply:
This is not for strings.
This is when you have a list of objects that contain references to other objects or lists.  This is the bit the serializes those objects and it was using the the alignment of the object in the top level list when serializing the referered to objects.
I moved the alignment code because in the case of a list it was not needed and the correct alignment at that point is that of the contents of the list and not 8.